src/user/utils/user.factory.ts

```diff
@@ -0,0 +1,29 @@
+import { Student } from '../domain/student';
+import { Teacher } from '../domain/teacher';
+
+export class UserFactory {
+  static createStudent(prismaStudent): Student {
+    const student: Student = {
+      id: prismaStudent.user_id,
+      name: prismaStudent.user.name,
+      contactEmail: prismaStudent.contact_email,
+      description: prismaStudent.description,
+      enrollment: prismaStudent.enrollment,
+      whatsapp: prismaStudent.whatsapp,
+      linkedin: prismaStudent.linkedin,
+      courseId: prismaStudent.course_id,
```

Comment:
Retorna o course com ID e nome: 

```suggestion
      courseId: prismaStudent.course_id,
      course: prismaStudent.course && {
        id: prismaStudent.course.id,
        name: prismaStudent.course.name,
      },
```